src/fft.rg

```diff
@@ -279,48 +589,133 @@ function fft.generate_fft_interface(itype, dtype)
     end
   end
 
-  __demand(__inline)
-  task iface.execute_plan(input : region(ispace(itype), dtype),
-                          output : region(ispace(itype), dtype),
-                          plan : region(ispace(int1d), iface.plan))
+
+  ----EXECUTE PLAN FUNCTIONS----
+
+  --Task to execute plan. Calls cufftExecZ2Z if in GPU mode and fftw_execute_dft if in CPU mode
+   __demand(__inline)
+  task iface.execute_plan(input : region(ispace(itype), dtype_in), output : region(ispace(itype), dtype_out), plan : region(ispace(int1d), iface.plan))
   where reads(input, plan), writes(output) do
-    var p = iface.get_plan(plan, true)
-    var input_base = get_base(rect_t(input.ispace.bounds), __physical(input)[0], __fields(input)[0])
-    var output_base = get_base(rect_t(output.ispace.bounds), __physical(output)[0], __fields(output)[0])
+    format.println("In iface.execute_plan...")
 
+    --Get pointer to plan
+    var p = iface.get_plan(plan, true) --task iface.get_plan(plan : region(ispace(int1d), iface.plan), check : bool) : &iface.plan
+
+    --Get pointers to input and output regions
+    var input_base = get_base_in(rect_t_in(input.ispace.bounds), __physical(input)[0], __fields(input)[0])
+    var output_base = get_base_out(rect_t_out(output.ispace.bounds), __physical(output)[0], __fields(output)[0])
+
+    --Check if we are in GPU or CPU mode
     var proc = get_executing_processor(__runtime())
+    format.println("execute_plan: TOC PROC IS {}",c.TOC_PROC) --TOC = Throughput Oriented Core: Means we have a GPU
+    format.println("execute_plan: Processor kind is {}", c.legion_processor_kind(proc))
+    format.println("size of dtype is {}", dtype_size) --dtype size is 8 for complex32 and 16 for complex64
+
+    --If in GPU mode, use cufftExec
     if c.legion_processor_kind(proc) == c.TOC_PROC then
       c.printf("execute plan via cuFFT\n")
+      var ok = 0
```

Comment:
Should we split this out to have a separate GPU task like the others?

Reply:
Yes, have split this out.